Scripts/Tracking/Follow/Modifier/FollowModifier.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace VRTK.Core.Tracking.Follow.Modifier
+{
+    using UnityEngine;
+
+    /// <summary>
+    /// The FollowModifier provides a way of describing logic to modify a Transform's position, rotation and scale.
+    /// </summary>
+    public abstract class FollowModifier : MonoBehaviour
+    {
+        protected Transform cachedSource;
+        protected Transform cachedTarget;
+
+        /// <summary>
+        /// The GetSource method returns the current source being used in the modifier process.
+        /// </summary>
+        /// <returns>A Transform of the current source.</returns>
+        public virtual Transform GetSource()

```

Comment:
We need to discuss this: I'd vote to use `get`-only properties instead of getter methods instead.

Reply:
personally i don't like `readonly` properties i find it confusing that you have a variable that you can't set

like when you have `transform.localScale = something` is fine but `transform.lossyScale = something` doesn't work.

i like variables to be variable :)

Is there something in the msdn guidelines that we could adhere to?